features/karaf/openhab-tp/src/main/feature/feature.xml

```diff
@@ -232,6 +232,15 @@
 		<bundle>mvn:org.eclipse.kura/org.eclipse.soda.dk.comm.x86_64/1.2.201</bundle>
 	</feature>
 
+	<conditional>
+		<condition>req:osgi.native;filter:="(&amp;(osgi.native.osname=MacOSX)(osgi.native.processor=AArch64))"</condition>
```

Comment:
It seems that `conditional` element can be specified under `<feature>` element to determine conditional dependencies. From this point of view proper place might be a upper level feature such as `openhab.tp-serial` (if one exists) which can make two conditions 1) for macos/aarch64 - which you made, calling feature `openhab.tp-serial-purejavacomm`, and 2) for anything else than macos/aarch64 pulling in `openhab.tp-serial-rxtx`.

Reply:
See for example:

https://github.com/openhab/openhab-core/blob/d3d0fe3995b2b950e253d782efbda33b31eebeec/features/karaf/openhab-core/src/main/feature/feature.xml#L523-L533